corehq/ex-submodules/dimagi/utils/couch/migration.py

```diff
@@ -84,6 +103,17 @@ def _migration_sync_to_sql(self, sql_object):
         for field_name in self._migration_get_fields():
             value = getattr(self, field_name)
             setattr(sql_object, field_name, value)
+        for spec in self._migration_get_submodels():
+            sql_submodels = []
+            for couch_submodel in getattr(self, spec.couch_attr):
+                sql_fields = {
+                    sql_field: getattr(couch_submodel, couch_field)
+                    for couch_field, sql_field in zip(spec.couch_fields, spec.sql_fields)
+                }
+                sql_submodels.append(spec.sql_class(**sql_fields))
+            sql_attr = getattr(sql_object, spec.sql_attr)
+            sql_attr.all().delete()
```

Comment:
won't this be very expensive? Is there no way to use `get_or_create` for the sql models?

Reply:
I think it'll add a fair amount of complexity, since there's no couch id for the sub models. I'm inclined to leave it as-is for now and revisit when there's a higher volume migration (or if the custom data fields migration turns out to be slow).